evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -964,6 +964,26 @@ class MockPipelineNone(BinaryClassificationPipeline):
     assert MockPipelineNone(parameters={}).hyperparameters == {'Mock Classifier': {}}
 
 
+def test_hyperparameters_linear_pipeline_duplicate_components():
+
+    class PipeLineLinear(BinaryClassificationPipeline):
+        component_graph = ["One Hot Encoder", "One Hot Encoder", "Random Forest Classifier"]
+
+    assert PipeLineLinear.hyperparameters == {'One Hot Encoder': {},
+                                              "One Hot Encoder_1": {},
+                                              'Random Forest Classifier': {'n_estimators': Integer(10, 1000),
+                                                                           'max_depth': Integer(1, 10)}}
+
+    class PipeLineLinear(BinaryClassificationPipeline):
+        custom_hyperparameters = {"One Hot Encoder_1": {"top_n": Integer(10, 50)}}
+        component_graph = ["One Hot Encoder", "One Hot Encoder", "Random Forest Classifier"]
```

Comment:
Same comment above about weirdness w/ user knowing to specify "One Hot Encoder_1"... maybe the right thing to do is just to have some documentation about the naming convention for duplicate components?

Reply:
Yea, I anticipate that's going to be a big gotcha going forward...